cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/validator/PreCompileValidatorImpl.java

```diff
@@ -238,37 +238,108 @@ private void validateFlow(Flow compiledFlow, ExecutableModellingResult result) {
         if (CollectionUtils.isEmpty(compiledFlow.getWorkflow().getSteps())) {
             result.getErrors().add(new RuntimeException("Flow: " + compiledFlow.getName() + " has no steps"));
         } else {
-            RuntimeException exception = validateNavigation(compiledFlow.getWorkflow().getSteps().getFirst(), compiledFlow.getWorkflow().getSteps(),
-                    getStepNames(compiledFlow));
-            if (exception != null) {
-                result.getErrors().add(exception);
-            }
+            List<RuntimeException> navigationErrors = new ArrayList<>();
+            Set<String> reachableStepNames = new HashSet<>();
+            Set<String> reachableResultNames = new HashSet<>();
+            Set<String> resultNames = getResultNames(compiledFlow);
+            Deque<Step> steps = compiledFlow.getWorkflow().getSteps();
+            List<RuntimeException> stepErrors = new ArrayList<>();
+            List<RuntimeException> resultsErrors = new ArrayList<>();
+
+            validateNavigation(
+                    compiledFlow.getWorkflow().getSteps().getFirst(),
+                    steps,
+                    resultNames,
+                    reachableStepNames,
+                    reachableResultNames,
+                    navigationErrors
```

Comment:
maybe you could put here `result.getErrors()` instead of navigationErrors, same thing for stepErrors and resultsErrors